ts/src/oxfun.ts

```diff
@@ -0,0 +1,2463 @@
+            'tag': undefined,
+            'tagTo': undefined,
+            'tagFrom': undefined,
+            'type': type,
+            'amount': amount,
+            'currency': code,
+            'status': status,
+            'updated': undefined,
+            'internal': undefined,
+            'comment': undefined,
+            'fee': fee,
+        };
+    }
+
+    parseDepositStatus (status) {
+        const statuses = {
+            'COMPLETED': 'ok',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    parseWithdrawalStatus (status) {
+        const statuses = {
+            'COMPLETED': 'ok',
+            'PROCESSING': 'pending',
+            'IN SWEEPING': 'pending',
+            'PENDING': 'pending',
+            'ON HOLD': 'pending',
+            'CANCELED': 'canceled',
+            'FAILED': 'failed',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    async withdraw (code: string, amount: number, address: string, tag = undefined, params = {}) { // todo does it work?
+        /**
+         * @method
+         * @name bitflex#withdraw
+         * @description make a withdrawal
+         * @see https://docs.bitflex.com/spot#withdraw
+         * @param {string} code unified currency code
+         * @param {float} amount the amount to withdraw
+         * @param {string} address the address to withdraw to
+         * @param {string} tag
+         * @param {string} params.network network for withdraw
+         * @param {bool} params.externalFee if false, then the fee is taken from the quantity, also with the burn fee for asset SOLO
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         *
+         * EXCHANGE SPECIFIC PARAMETERS
+         * @param {string} [params.tfaType] GOOGLE, or AUTHY_SECRET, or YUBIKEY, for 2FA
+         * @param {string} [params.code] 2FA code
+         * @returns {object} a [transaction structure]{@link https://docs.ccxt.com/#/?id=transaction-structure}
+         */
+        [ tag, params ] = this.handleWithdrawTagAndParams (tag, params);
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const stringAmount = this.currencyToPrecision (code, amount);
+        const request = {
+            'asset': currency['id'],
+            'address': address,
+            'quantity': stringAmount,
+        };
+        if (tag !== undefined) {
+            request['memo'] = tag;
+        }
+        let networkCode = undefined;
+        [ networkCode, params ] = this.handleNetworkCodeAndParams (params);
+        if (networkCode !== undefined) {
+            request['network'] = this.networkCodeToId (networkCode);
+        }
+        let externalFee = false;
+        if ('externalFee' in params) {
+            externalFee = this.safeBool (params, 'externalFee');
+            params = this.omit (params, 'externalFee');
+        }
+        request['externalFee'] = externalFee;
```

Comment:
you don't need all of this logic, only `request['externalValue'] = false` (if the user provides it in params it will be overridden)

Reply:
ok